Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/StartRide/Internal.hs

```diff
@@ -24,17 +24,38 @@ import Kernel.External.Maps.Types (LatLong)
 import qualified Kernel.Storage.Esqueleto as Esq
 import Kernel.Storage.Esqueleto.Config (EsqLocDBFlow)
 import Kernel.Storage.Esqueleto.Transactionable (runInLocationDB)
+import Kernel.Streaming.Kafka.Producer.Types (KafkaProducerTools)
 import Kernel.Types.Common
 import Kernel.Types.Id
+import Kernel.Utils.Common
+import Lib.SessionizerMetrics.Types.Event
 import qualified SharedLogic.Ride as CQRide
 import Storage.CachedQueries.CacheConfig (CacheFlow)
 import qualified Storage.Queries.BusinessEvent as QBE
 import qualified Storage.Queries.Driver.DriverFlowStatus as QDFS
 import qualified Storage.Queries.DriverLocation as DrLoc
 import qualified Storage.Queries.Ride as QRide
+import Tools.Error
+import Tools.Event
 
-startRideTransaction :: (CacheFlow m r, EsqDBFlow m r, EsqLocDBFlow m r) => Id SP.Person -> Id SRide.Ride -> Id SRB.Booking -> LatLong -> Id Dmerch.Merchant -> m ()
+startRideTransaction ::
+  ( CacheFlow m r,
+    EsqDBFlow m r,
+    EsqLocDBFlow m r,
+    HasFlowEnv m r '["kafkaProducerTools" ::: KafkaProducerTools],
+    HasField "version" r r0,
+    HasField "getDeploymentVersion" r0 Text,
+    HasField "eventStreamMap" r [EventStreamMap]
+  ) =>
+  Id SP.Person ->
+  Id SRide.Ride ->
+  Id SRB.Booking ->
+  LatLong ->
+  Id Dmerch.Merchant ->
+  m ()
 startRideTransaction driverId rideId bookingId firstPoint merchantId = do
+  ride <- QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
```

Comment:
the same, you can pass ride from external code

Reply:
Done